common/types/client.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+import (
+	"context"
+	"math/big"
+)
+
+type Client[CHAINID any, ADDR comparable, BLOCK any, TX any, HASH comparable, TXRECEIPT any, EVENT any, EVENTOPS any] interface {
```

Comment:
Add comments explaining this interface.
// Client is a generic interface which ALL chains can use, to talk to RPC endpoints for their chain, and receive a generic response. Every native chain code will implement this interface internally.